.github/workflows/package_installers.yml

```diff
@@ -33,35 +33,83 @@ jobs:
           # We need to download all tags so that the axion-release-plugin
           # can resolve the most recent version tag.
           fetch-depth: 0
+
       - name: Set up JDK 17
         uses: actions/setup-java@v2
         with:
           # We need a recent version of Java with jpackage included.
           java-version: '17'
           # We use the zulu distribution, which is an OpenJDK distro.
           distribution: 'zulu'
-      - name: Package GUI app installer with Gradle
+
+      # We create a code-signing keychain on MacOS before building and packaging the app, as the
+      # app will be signed as part of the jpackage build phase.
+      - name: "MacOS - Import Certificate: Developer ID Application"
+        if: matrix.os == 'macos-latest'
+        uses: devbotsxyz/import-signing-certificate@main
+        with:
+          # This should be a certificate + private key, exported as a p12 file and base64 encoded as
+          # a GitHub secret.  The certificate should be the:
+          # 'Developer ID Application: The International Data Organization For Transport (BF2U75HN4D)'
+          # certificate, locked with the specified passphrase.
+          certificate-data: ${{ secrets.DEVELOPER_ID_APPLICATION_CERTIFICATE_P12_BASE64 }}
+          certificate-passphrase: ${{ secrets.DEVELOPER_ID_APPLICATION_CERTIFICATE_PASSWORD }}
+          # The resulting keychain will be locked with this separate password.
+          keychain-password: ${{ secrets.KEYCHAIN_PASSWORD }}
```

Comment:
Code-signing is a cheap operation (compare to notarization, as discussed below) but it still requires access to secrets.  We could restrict codesigning to just pushes+releases on the main master branch so that PRs would still run successfully?

Reply:
I think it's fair to restrict codesigning and notarization to just pushes+releases. `README.md` refers users to the release page to download the app, so I think most users won't need the PR app, only the contributors.